### PR TITLE
Add option to override SOURCE_DATE_EPOCH.

### PR DESCRIPTION
**Pull request by zaclegarssure**
`flatpak-builder` overrides `SOURCE_DATE_EPOCH` with the last modification time of the manifest file (of the program being built).
With this option, it would provide a more convenient interface to perform the same thing, useful when trying to reproduce a build.
Concretely I just added an optional parameter `--override-source-date-epoch` which, if set, will override the `SOURCE_DATE_EPOCH` with the given value. 
It's my first time looking at the code of this project, therefore, even though I made minimal changes, I might have made some mistakes.
Update:
I also added the timestamp to the resolved manifest, to actually have reproducible builds.

### CODE REVIEW

File: src/builder-manifest.c [hunk 1]
```diff
@@ -1090,6 +1100,16 @@ builder_manifest_class_init (BuilderManifestClass *klass)
...
+
+  g_object_class_install_property (object_class,
+                                   PROP_SOURCE_DATE_EPOCH,
+                                   g_param_spec_int64 ("source-date-epoch",
+                                                     "",
+                                                     "",
+                                                     0,
+                                                     G_MAXINT64,
+                                                     0,
+                                                     G_PARAM_READWRITE));
```
Comment (TingPing) [inline src/builder-manifest.c:1112]:
```suggestion
                                                      "",
                                                      "",
                                                      0,
                                                      G_MAXINT64,
                                                      0,
                                                      G_PARAM_READWRITE));
```

Comment (TingPing) [reply src/builder-manifest.c:1112]: Hard to tell on the github UI, but alignment was just off by one or two.

File: src/builder-manifest.c [hunk 2]
```diff
@@ -1629,6 +1649,8 @@ builder_manifest_start (BuilderManifest *self,
   g_autofree char *sdk_path = NULL;
   const char *stop_at;
 
+  self->source_date_epoch = builder_context_get_source_date_epoch(context);
```
Comment (TingPing) [inline src/builder-manifest.c:1652]:
```suggestion
  self->source_date_epoch = builder_context_get_source_date_epoch (context);
```

**Comment by TingPing** [comment]
The change itself sounds and looks good to me.

File: src/builder-manifest.c (hunk 1 above)
Comment (TingPing) [reply src/builder-manifest.c]: It was off by one more space sorry :/

Comment (zaclegarssure) [reply src/builder-manifest.c]: No problems, I should have been more careful in the first place.

**Pull request merged**